### PR TITLE
dashboard beginnings

### LINKED ISSUE #1962: design/prototype Dashboard to visually represent "it's working" or "it's broken"
From  [maint/tech debt storytime discussion Wed 10/19](https://docs.google.com/document/d/1vTv9ySljSgJ-zcrH2ZztDpdUtnVCxe_hgfZnIehkPGs/edit#heading=h.a6kqlslw7c5o):

Preservation system:

- What are metrics ?
- What does success look like?
- How can we have a visual representation of “yes it's working” or “no, it’s not working” for this complex system?
    - Would like an improvement over Honeybadger alerts.

- There are dashboards for google books and for techmd service.
- See the [Wiki page](https://github.com/sul-dlss/preservation_catalog/wiki/Notes-from-Preservation-Catalog-knowledge-transfer) that includes, "What might a user want to do manually?":
- Lots of useful queries in [DB readme](https://github.com/sul-dlss/preservation_catalog/blob/main/db/README.md)
    - Query the catalog to see what state a particular object or set of objects is in
    - how many objects are in a given state for a given storage root, 
    - how many objects are in a given state for a given replication endpoint, etc.
     - Integrity check individual objects, lists of objects, objects on a given storage root, etc.

@jmartin-sul will talk to Astrid and to Andrew about this.

-----
Motivation:   preservation has lots of pieces that all have to fit together:  cron jobs, code bases, background jobs, robots, prescat, external services, etc.  It’s a big, complex ecosystem that is unfamiliar to some.

-----
Exploratory prototype code is also welcome, but don't get too attached to it.

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made? 🤔

Fixes #1962 

See https://preservation-catalog-web-stage-01.stanford.edu/dashboard

This is, essentially, some ideas with first pass implemetations.  I would expect:
- more, better ideas to come
- better display of information
- potential implementation improvements

Once this PR is merged, we can have multiple folks simultaneously work on improving the PR.  So it is less important that this code is close to perfect, and more important that the functionality has tests, as lots of refactors are expected.

There are already a bunch of tickets with label dashbaord for improvements;  I am about to make more from the notes from storytime last week.

#### Catalog Status clicked:

image

#### Replication Status clicked:

image

#### Audit Status clicked:

image

## How was this change tested? 🤨

specs and in situ

⚡ ⚠ If this change has cross service impact, or if it changes code used internally for cloud replication, ***run [integration test create_preassembly_image_spec.rb](https://github.com/sul-dlss/infrastructure-integration-test/blob/main/spec/features/create_preassembly_image_spec.rb) on stage as it tests preservation***, and/or test in stage environment, in addition to specs. The main classes relevant to replication are `ZipmakerJob`, `PlexerJob`, `*DeliveryJob`, `ResultsRecorderJob`, and `DruidVersionZip`; [see here for overview diagram of replication pipeline](https://github.com/sul-dlss/preservation_catalog/blob/main/app/jobs/README.md).⚡

### CODE REVIEW

**Comment by ndushay** [comment]
build failure will be fixed by #2015;  the lint failure is not related to this code.

**Comment by ndushay** [comment]
> Quick question in advance of digging in: what is the current approach for authorization? Rely on IP-based/firewalling?

Whatever it is we do for `/resque`

File: spec/helpers/dashboard_helper_spec.rb [hunk 1]
```diff
@@ -0,0 +1,677 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe DashboardHelper do
```
Comment (ndushay) [inline spec/helpers/dashboard_helper_spec.rb:5]: the meat of the logic is in the helper class, so this is where most of the spec code is.

File: spec/routing/routing_spec.rb [hunk 2]
```diff
@@ -38,4 +38,10 @@
       expect(post: content_diff_object_path(id: id)).to route_to(controller: 'objects', action: 'content_diff', id: id)
     end
   end
+
+  describe 'dashboard' do
+    it 'routes to dashboard controller index action' do
+      expect(get: dashboard_path).to route_to(controller: 'dashboard', action: 'index', format: 'html')
+    end
+  end
```
Comment (ndushay) [inline spec/routing/routing_spec.rb:46]: I mostly did this to get the route set up properly ... but eh, why not keep it.

**Review by ndushay — COMMENTED** [review]
The views are organized like this:

Once this PR is merged, we can have multiple folks simultaneously work on improving the PR.  So it is less important that this code is close to perfect, and more important that the functionality has tests, as lots of refactors are expected.

index - has the bootstrap navbar html
_xxx_status - in the navbar, where xxx is catalog, replication or audit
_xxx_information - in the body, ditto xxx

and the other partials are used by xxx_information

**Review by aaron-collier — APPROVED** [review]

**Comment by ndushay** [comment]
John said this in slack, so I'm merging to unblock other work:

>looking now, but no objection to merging without waiting for me -- shouldn't break any existing functionality, and i'm happy to make changes in follow-on PRs for stuff that might need to be revistied/refined.

**Pull request merged**